### PR TITLE
Introduce pinot-plugin assembly-desciptor as a better alternative tha…

### PR DESCRIPTION
**Pull request by rfscholte**
…n shading

Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
   3. `performance`
   4. `ui`
   5. `backward-incompat`
   6. `release-notes` (**)
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `dependencies`
- `docker`
- `kubernetes`
- `observability`
- `security`
- `code-style`
- `extension-point`
- `refactor`
- `cleanup`

(**) Use `release-notes` label for scenarios like:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed

### CODE REVIEW

**Comment by KKcorps** [comment]
@rfscholte Thanks for this change! How does it tackle the case where two classes have same package and names in the zip?

**Comment by gortiz** [comment]
please run mvn license:format in assembly-descriptor to add the license to the two files that do not include it

**Review by gortiz — APPROVED** [review]

File: pinot-plugins/assembly-descriptor/src/it/projects/simple-assembly/src/main/resources/pinot-plugin.properties [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+importFrom.pinot=org.apache.pinot.segment.spi
```
Comment (gortiz) [inline pinot-plugins/assembly-descriptor/src/it/projects/simple-assembly/src/main/resources/pinot-plugin.properties:20]: This is just an example, right? AFAIR that package is already imported by default

Comment (rfscholte) [reply pinot-plugins/assembly-descriptor/src/it/projects/simple-assembly/src/main/resources/pinot-plugin.properties:20]: Yes, this is just an example, but also to proof that it works.
And no, this package it currently imported by default, notice the word `segment`

Comment (rfscholte) [reply pinot-plugins/assembly-descriptor/src/it/projects/simple-assembly/src/main/resources/pinot-plugin.properties:20]: we could change it to any other package if that makes the example more clear

**Pull request merged**